### PR TITLE
Fix load supported setups

### PR DESCRIPTION
**Pull request by gbe**
As reported in https://github.com/ThePorgs/Exegol-images/issues/207, this PR fixes the issue introduced when the variables were renamed from lowercase to uppercase.

One of the variable was `path` and declared in a function, hence renaming to PATH overrode the environment variable.

This PR:
- fixes the issue by renaming the PATH variable to lpath
- all local variables are now explicitely scoped with the `local` keyword, and renamed in lowercase, leaving the UPPERCASE variable to ones with a global scope
- all local variables start with the lowercase letter `l`
- single square brackets removed by double square brackets to avoid any issues.

### CODE REVIEW

**Review by ShutdownRepo — CHANGES_REQUESTED** [review]
Awesome, thank you for these changes. Two questions/requests before merging:
- Shouldn't we switch to `zsh` in the shebang?
- I think it'd be best to keep the `local` scope and remove the `l` prepend in the variables name

**Comment by gbe** [comment]
- I removed the prefixed lowercase letter L, as requested
- Shell interpreter is switched to `zsh`. To do so, I had to adapt the deployment of `apt` and `firefox addons` as they were either failing or printing error messages.
- Little extra: there were some temporary folders and files, I switched to a dynamic approach with `mktemp`.

**Review by ShutdownRepo — CHANGES_REQUESTED** [review]
Last change/question, and then we'll be good to go

File: sources/assets/exegol/load_supported_setups.sh
Comment (gbe) [reply sources/assets/exegol/load_supported_setups.sh:8]: No, I left it on purpose. Looking at the commit you gave, the aliases are indeed loaded in the zshrc. However, the zshrc is automatically loaded from within an interactive session. It's not the case here as it is a script.
It's therefore required to explicitely source $EXEGOL_ALIASES.
Plus, I had tested without it, bqm still fails to be found.

Comment (ShutdownRepo) [reply sources/assets/exegol/load_supported_setups.sh:8]: Then would it be possible to replace this block with a `source ~/.zshrc` ?

Comment (ShutdownRepo) [reply sources/assets/exegol/load_supported_setups.sh:8]: This will allow the script to follow the changes made to zhsrc in the future

Comment (gbe) [reply sources/assets/exegol/load_supported_setups.sh:8]: I see your point, but I discourage this approach as it could introduce some side-effects in the script execution depending on what's added into zshrc. A side-effect that would go unnoticed as we've seen as there's currently no check on this file when running the pipeline. This is why the scope extension should be on a need-basis to avoid any issues.
I believe it would be a lot easier to troubleshoot a missing source aliasesfile than any funky commands/variables from a zshrc.
If you absolutely want me to source ~/.zshrc I'll do it though.

Comment (ShutdownRepo) [reply sources/assets/exegol/load_supported_setups.sh:8]: Hmm, I see your point, we'll go as-is, and in the future, if we make any check of the zshrc content, or if we want to switch to what I described for some reason, we will. However, while I agree with you on the side-effects, there are benefits to sourcing zhsrc. One example is what we have with pyenv. Pyenv is set up in the zshrc, but not in the load_supported_setups, preventing python/pip from working correctly (because it's not in the PATH).
The example with pyenv could be encountered with other things that are set up in zshrc and not elsewhere, and sourcing zshrc would allow to not have duplicate code hanging around

**Comment by ShutdownRepo** [comment]
Running pipeline, will merge if successful

File: sources/assets/exegol/load_supported_setups.sh
Comment (gbe) [reply sources/assets/exegol/load_supported_setups.sh:8]: Yep, valid point, sometimes the benefits are worth taking some risks ;), and indeed it could be more interesting to switch.

**Pull request merged**
